src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -675,19 +693,23 @@ public static void ReportParameterErrors(
                 // error CS1100: Method '{0}' has a parameter modifier 'this' which is not on the first parameter
                 diagnostics.Add(ErrorCode.ERR_BadThisParam, thisKeyword.GetLocation(), owner?.Name ?? "");
             }
-            else if (isParams && owner is { } && owner.IsOperator())
+            else if (isParams && owner is { } && (owner.IsOperator() || inExtension))
             {
                 // error CS1670: params is not valid in this context
                 diagnostics.Add(ErrorCode.ERR_IllegalParams, paramsKeyword.GetLocation());
             }
             else if (!typeWithAnnotations.IsDefault && typeWithAnnotations.IsStatic)
             {
-                Debug.Assert(containingSymbol is null || (containingSymbol is FunctionPointerMethodSymbol or { ContainingType: not null }));
-                // error CS0721: '{0}': static types cannot be used as parameters
-                diagnostics.Add(
-                    ErrorFacts.GetStaticClassParameterCode(containingSymbol?.ContainingType?.IsInterfaceType() ?? false),
-                    syntax.Type?.Location ?? syntax.GetLocation(),
-                    typeWithAnnotations.Type);
+                Debug.Assert(containingSymbol is null || (containingSymbol is FunctionPointerMethodSymbol or { ContainingType: not null }) || inExtension);
+
+                if (!inExtension || isNamed)
```

Comment:
>isNamed [](http://example.com/codeflow?start=36&length=7)

It is not obvious why presence of the name should make a difference.

Reply:
If you write `extension(Type x)`, it means we'll have an instance `x`, so the type may not be static.
If you write `extension(Type)`, you may be extending a static or an instance type (there will not be a `this` parameter holding an instance of this type). `extension(StaticType)` should not produce an error.